src/Controller/LoginController.php

```diff
@@ -0,0 +1,107 @@
+    {
+        $authenticationProvider = null;
+
+        // Look for authentication provider (aliased to "role") in query string.
+        if (null !== $request->query->has('role')) {
+            $authenticationProvider = $request->query->get('role');
+        }
+
+        // Look for authentication provider (aliased to "role") in target path query string.
+        if (null === $authenticationProvider) {
+            $targetPath = $this->getTargetPath($request->getSession(), 'main');
+            if (null !== $targetPath) {
+                $queryString = parse_url($targetPath, PHP_URL_QUERY);
+                if (null !== $queryString) {
+                    parse_str($queryString, $targetQuery);
+                    $authenticationProvider = $targetQuery['role'] ?? null;
+                }
+            }
+        }
+
+        // Get authentication provider from cookie.
+        if (null === $authenticationProvider) {
+            $authenticationProviderCookieName = 'authentication_provider';
+            if (null === $request->get('reset-authentication-provider')) {
+                $authenticationProvider = $request->cookies->get($authenticationProviderCookieName);
+            }
+        }
+
+        $rememberProvider = false;
+
+        $formBuilder = $this->createFormBuilder();
+        foreach ($this->providerManager->getProviderKeys() as $key) {
+            $name = 'open_id_connect.login_provider.'.$key;
+            $formBuilder->add('provider_'.$key, SubmitType::class, [
+                'label' => new TranslatableMessage($name, [], 'login'),
+            ]);
+        }
+        $form = $formBuilder
+            ->add('remember_provider', CheckboxType::class, [
+                'label' => new TranslatableMessage('Remember my choice', [], 'login'),
+                'required' => false,
+            ])
+            ->getForm()
+        ;
+
+        $form->handleRequest($request);
+        if ($form->isSubmitted() && $form->isValid()) {
+            foreach ($this->providerManager->getProviderKeys() as $key) {
+                if ($form->get('provider_'.$key)->isClicked()) {
+                    $authenticationProvider = $key;
+                    break;
+                }
+            }
+            $rememberProvider = $form->get('remember_provider')->getData();
+        }
+
+        if (null !== $authenticationProvider) {
+            try {
+                $this->providerManager->getProvider($authenticationProvider);
+                $response = $this->redirectToRoute('itkdev_openid_connect_login',
+                    ['providerKey' => $authenticationProvider]);
+                if ($rememberProvider) {
+                    $response->headers->setCookie(new Cookie($authenticationProviderCookieName, $authenticationProvider, new \DateTimeImmutable('+1 month')));
+                }
+
+                return $response;
+            } catch (InvalidProviderException $invalidProviderException) {
+            }
```

Comment:
Maybe a comment about why no meaningful error message can be provided to the user.

Reply:
Fixed in https://github.com/itk-dev/naevnssekretariatet/pull/75/commits/5db766241a47a02c9c084f29967348c60539e756